### PR TITLE
Updated power_of_4.py

### PR DESCRIPTION
**Pull request by AseemGupta39**

### CODE REVIEW

**Review by imSanko — COMMENTED** [review]
Check with the `pre-commit run --all-files --show-diff-on-failure` for your errors to go through

File: bit_manipulation/power_of_4.py [hunk 1]
```diff
@@ -52,13 +52,9 @@ def power_of_4(number: int) -> bool:
...
         raise ValueError("number must be positive")
     if number & (number - 1) == 0:
-        c = 0
-        while number:
-            c += 1
-            number >>= 1
-        return c % 2 == 1
-    else:
-        return False
+        while number != 1 and number != 2:
```
Comment (cclauss) [inline bit_manipulation/power_of_4.py:55]:
```suggestion
        while number not in (1, 2):
```

File: bit_manipulation/power_of_4.py [hunk 2]
```diff
@@ -52,13 +52,9 @@ def power_of_4(number: int) -> bool:
     if number <= 0:
         raise ValueError("number must be positive")
     if number & (number - 1) == 0:
-        c = 0
-        while number:
-            c += 1
-            number >>= 1
-        return c % 2 == 1
-    else:
-        return False
```
Comment (tianyizheng02) [inline bit_manipulation/power_of_4.py:61]: The function now doesn't return anything if the if-statement evaluates to False. This is why the build is failing.

**Review by tianyizheng02 — CHANGES_REQUESTED** [review]

**Comment by cclauss** [comment]
Closing tests_are_failing PRs to prepare for Hacktoberfest 2024

**Pull request closed without merging**